### ISSUE TITLE
(eslint-plugin-template) Rule proposal: Equivalent of eslint lint rule `prefer-template`

### ISSUE THREAD

**Issue by g-drouard** [angular-eslint/angular-eslint#2289]
Equivalent of [prefer-template](https://eslint.org/docs/latest/rules/prefer-template) in template files.

With Angular 19.2 template literals expressions are supported : https://github.com/angular/angular/releases/tag/19.2.0

❌ Invalid Code :

```html
<h1>{{ 'translate-prefix-' + translateKey | translate }}</h1>
<my-component [class]="'my-class-prefix-' + myClass"
```

✅ Valid Code :

```html
<h1>{{ `translate-prefix-${translateKey}` | translate }}</h1>
<my-component [class]="`my-class-prefix-${myClass}`"
```

**Comment by JamesHenry** [comment]
Cool, thanks, makes sense to have a rule for this, PRs welcome!

`@angular-eslint/template/prefer-template` is probably not the most clear and obvious name given the overloading of the word "template" in the context of Angular.

Maybe (even though a little silly) `@angular-eslint/template/prefer-template-in-template` is a bit clearer, WDYT?
